modules/theme-tailwind-magento/molecules/MiniCartItem.vue

```diff
@@ -0,0 +1,163 @@
+            </h4>
+          </div>
+        </div>
+      </div>
+      <div class="col-span-2 mt-2 flex items-center justify-between">
+        <p class="flex flex-col mt-1 text-sm font-medium text-gray-900 flex-end text-right">
+          <SfxMoney :money="cartItem.rowTotal" el="strong" class="text-lg" />
+        </p>
+      </div>
+      <div class="col-span-3 mt-2 flex items-center justify-between">
+        <div class="flex gap-2">
+          <Button @click="onDec">-</Button>
+          <Input :value="tmpQuantity" :inputmode="'decimal'" class="w-16 text-center" @input="onInput" />
+          <Button @click="onInc">+</Button>
+        </div>
+      </div>
+
+      <div class="col-span-1 flex items-center justify-center">
+        <a class="text-gray-400 hover:text-red-500 hover:cursor-pointer" @click="promptRemoval">
+          <SolidTrash />
+        </a>
+      </div>
+    </div>
+
+    <ConfirmProductRemovalModal
+      v-if="isRemoveModalOpen"
+      :name="cartItem.product.name"
+      @close="keepItem"
+      @keep-product="keepItem"
+      @remove-product="removeItem"
+    />
+  </li>
+</template>
+
+<script setup lang="ts">
+import SfxMoney from '#ioc/components/SfxMoney'
+import useCartItem from '#ioc/composables/useCartItem'
+import ToCartItem from '#ioc/mappers/ToCartItem'
+import resizeImage from '#ioc/utils/url/resizeImage'
+import debounce from '#ioc/utils/debounce'
+import SolidTrash from '#ioc/icons/SolidTrash'
+import Button from '#ioc/atoms/Button'
+import Input from '#ioc/atoms/Input'
+import { computed, defineAsyncComponent, PropType, ref, toRef, watch } from 'vue'
+import useRemoveFromCart from '#ioc/services/useRemoveFromCart'
+import useShowErrorNotification from '#ioc/composables/useShowErrorNotification'
+import useUpdateCartItem from '#ioc/services/useUpdateCartItem'
+
+const ConfirmProductRemovalModal = defineAsyncComponent(() => import('#ioc/organisms/ConfirmProductRemovalModal'))
+
+const props = defineProps({
+  cartItem: {
+    type: Object as PropType<ReturnType<typeof ToCartItem>>,
+    required: true,
+  },
+})
+
+const cartItem = useCartItem(toRef(props, 'cartItem'))
+const removeFromCart = useRemoveFromCart()
+const updateCartItem = useUpdateCartItem()
+const showErrorNotification = useShowErrorNotification()
+
+const tmpQuantity = ref(props.cartItem.quantity)
+const isRemoveModalOpen = ref(false)
+const isRemoveLoading = ref(false)
+
+const src = computed(() =>
+  resizeImage({
+    w: 80,
+    h: 80,
+    path: cartItem.product.thumbnailUrl,
+    fit: 'contain',
+  }),
+)
+
+watch(
+  tmpQuantity,
+  debounce(async (value: number) => {
+    if (value === 0) {
+      await promptRemoval()
+    } else if (value > 0) {
+      await updateQuantity(value)
+    } else {
+      tmpQuantity.value = cartItem.quantity
+    }
+  }, 250),
+)
+
+const onInput = (e: any) => {
+  if (e.target.quantity === '') {
+    return
+  } else {
+    tmpQuantity.value = Number(e.target.value)
+  }
+}
+
+const onInc = () => {
```

Comment:
Extract to useCartItem and reuse also in cart item? Now this logic is duplicated with cart item.